src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace System.Formats.Tar
+{
+    // Describes the header attributes from a tar archive entry.
+    // Supported formats:
+    // - 1979 Version 7 AT&T Unix Tar Command Format (v7).
+    // - POSIX IEEE 1003.1-1988 Unix Standard Tar Format (ustar).
+    // - POSIX IEEE 1003.1-2001 ("POSIX.1") Pax Interchange Tar Format (pax).
+    // - GNU Tar Format (gnu).
+    // Documentation: https://www.freebsd.org/cgi/man.cgi?query=tar&sektion=5
+    internal partial struct TarHeader
```

Comment:
this `struct` seems to be very large and I would not expect it to get us some perf. What are we getting by keeping it a struct?

Reply:
Would you prefer if I use class? And maybe split it into `internal class TarHeaderRead` and `internal class TarHeaderWrite`?

Reply:
I expect us to get no perf benefits from keeping it `struct`, while we can run into issues like modifying a copy instead:


```cs
TarHeader header = new();
header._mode = 1;

SetMode(header);

Console.WriteLine(header._mode); // prints 1 as we have passed struct copy to SetMode and modified a copy

static void SetMode(TarHeader h) => h._mode = 2;